### PR TITLE
[SPARK-43028][SQL] Add error class SQL_CONF_NOT_FOUND 

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

This PR adds a new error class `SQL_CONF_NOT_FOUND`.

### Why are the changes needed?

To make the error message more user-friendly when getting a non-existing SQL config. For example:
```
spark.conf.get("some.conf")
```
Before this PR, it will throw this error:
```
java.util.NoSuchElementException: some.conf
```
After this PR:
```
[SQL_CONF_NOT_FOUND] The SQL config "some.conf" cannot be found. Please verify that the config exists.
```

### Does this PR introduce _any_ user-facing change?

Yes. The error message will be changed.

### How was this patch tested?

Added a new UT.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1414,6 +1414,11 @@
       "sortBy must be used together with bucketBy."
     ]
   },
+  "SQL_CONF_NOT_FOUND" : {
+    "message" : [
+      "The SQL config \"<key>\" cannot be found. Please verify that the config exists."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1419]: Please, quote the config in the code and remove "" around the key:
```suggestion
      "The SQL config <sqlConf> cannot be found. Please verify that the config exists."
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 2]
```diff
@@ -2134,8 +2134,10 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
       cause = null)
   }
 
-  def noSuchElementExceptionError(key: String): Throwable = {
-    new NoSuchElementException(key)
+  def sqlConfigNotFoundError(key: String): SparkRuntimeException = {
+    new SparkRuntimeException(
+      errorClass = "SQL_CONF_NOT_FOUND",
+      messageParameters = Map("key" -> key))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:2140]: Please, quote the config using `toSQLConf()`

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by allisonwang-db** [comment]
cc @srielau

**Comment by MaxGekk** [comment]
@allisonwang-db Could you fix the R test:
```
══ Failed ══════════════════════════════════════════════════════════════════════
── 1. Failure ('test_sparkSQL.R:3963'): Setting and getting config on SparkSessi
`sparkR.conf("completely.dummy")` threw an error with unexpected message.
Expected match: "Config 'completely.dummy' is not set"
Actual message: "Unknown error: Error in handleErrors(returnStatus, conn): org.apache.spark.SparkNoSuchElementException: 
```

**Review by srielau — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @allisonwang-db and @srielau for review.

**Pull request closed without merging**